### DISCUSSION TITLE
[FEAT] Folding from inside the list

### DISCUSSION THREAD

**Discussion by filyp** [vslinko/obsidian-outliner#150] [Ideas]
Right now, a list can be folded only when having the cursor at the parent item.

It would be neat, if when trying to fold from some child, it would move the cursor to the parent and fold. Of course this child can also have it's own children, so they would get priority to be folded, but if they're already folded, it would fold the parent.

**Comment by vslinko** [comment]
@fsondej It sounds like this may be misleading behavior. Could you explain your use case when you need this functionality?

**Reply by filyp** [reply]
For example when I want to fold a list after editing some child, now I have to move back to the parent and then fold. It's a minor thing, but I do it often, because I have many nested lists that I keep folded usually.

I find it quite intuitive, for example it's how tree view in Dolphin file explorer works, and I intuitively expected to find the same behavior here.

Also I don't see a scenario where this behavior would conflict with some other intended behavior. Right now, in this scenario I described nothing happens. Someone will try to fold from inside of the list only if they expect this list to get folded.

**Reply by filyp** [reply]
So what do you think?

**Reply by imed-ghomari** [reply]
Being wanting this feature for some times too, I would use it a lot.

**Reply by vslinko** [reply]
I keep this idea for the future. I think I could implement it, but I think it should be under feature toggle.

**Reply by imed-ghomari** [reply]
Yeah, sure as you wish, as long as it implemented I don't mind.

**Reply by blackeyeliner** [reply]
Notion allows structures list navigation as described, and this behavior allows super fast navigation across structured lists. Thinking about which item I am on, can I collapse it or not, choosing that line and only then collapsing does not allow automatic operation - I have to pause, consider, etc.
